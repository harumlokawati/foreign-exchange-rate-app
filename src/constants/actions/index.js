export const FAILURE = 'FAILURE';
export const LOAD_RATE = 'LOAD_RATE';
export const LOAD_RATE_SUCCESS = 'LOAD_RATE_SUCCESS';
export const SET_AMOUNT = 'SET_AMOUNT';
export const SET_BASE = 'SET_BASE'
export const ADD_SYMBOL = 'ADD_SYMBOL';
export const REMOVE_SYMBOL = 'REMOVE_SYMBOL';
