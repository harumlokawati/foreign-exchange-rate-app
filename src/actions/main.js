import * as actions from '../constants/actions';

export function failure(error) {
    return {
        type: actions.FAILURE,
        payload: {error}
    }
}

export function loadRate(symbols, base) {
    return {
        type: actions.LOAD_RATE,
        payload: {symbols, base}
    }
}

export function loadRateSuccess(data) {
    return {
        type: actions.LOAD_RATE_SUCCESS,
        payload: {data}
    }
}

export function setAmount(amount) {
    return {
        type: actions.SET_AMOUNT,
        payload: {amount}
    }
}

export function setBase(base) {
    return {
        type: actions.SET_BASE,
        payload: {base}
    }
}


export function addSymbol(symbol) {
    return {
        type: actions.ADD_SYMBOL,
        payload: {symbol}
    }
}

export function removeSymbol(symbol) {
    return {
        type: actions.REMOVE_SYMBOL,
        payload: {symbol}
    }
}

export function setLoading(loading) {
    return {
        type: actions.SET_LOADING,
        payload: {loading}
    }
}

