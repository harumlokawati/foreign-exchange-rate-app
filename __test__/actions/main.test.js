import configureStore from 'redux-mock-store';
import * as mainAction from '../../src/actions/main'

const mockStore = configureStore();
const store = mockStore();

describe('main actions', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions()
    })

    describe('load rate', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.loadRate(['EUR'], 'USD'))
            expect(store.getActions()).toMatchSnapshot()
        })
    })

    describe('load rate success', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.loadRateSuccess({
                "base": "EUR",
                "rates": {"USD": 1.1187, "GBP": 0.88318},
                "date": "2019-05-24"
            }))
            expect(store.getActions()).toMatchSnapshot()
        })
    })

    describe('set amount', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.setAmount(105))
            expect(store.getActions()).toMatchSnapshot()
        })
    })

    describe('set base', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.setBase('EUR'))
            expect(store.getActions()).toMatchSnapshot()
        })
    })

    describe('set add symbol', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.addSymbol('IDR'))
            expect(store.getActions()).toMatchSnapshot()
        })
    })

    describe('set remove symbol', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.removeSymbol('IDR'))
            expect(store.getActions()).toMatchSnapshot()
        })
    })


    describe('set loading', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(mainAction.setLoading(true))
            expect(store.getActions()).toMatchSnapshot()
        })
    })


})