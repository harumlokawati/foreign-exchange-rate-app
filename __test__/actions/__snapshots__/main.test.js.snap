// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main actions load rate Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "base": "USD",
      "symbols": Array [
        "EUR",
      ],
    },
    "type": "LOAD_RATE",
  },
]
`;

exports[`main actions load rate success Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Object {
        "base": "EUR",
        "date": "2019-05-24",
        "rates": Object {
          "GBP": 0.88318,
          "USD": 1.1187,
        },
      },
    },
    "type": "LOAD_RATE_SUCCESS",
  },
]
`;

exports[`main actions set add symbol Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "symbol": "IDR",
    },
    "type": "ADD_SYMBOL",
  },
]
`;

exports[`main actions set amount Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "amount": 105,
    },
    "type": "SET_AMOUNT",
  },
]
`;

exports[`main actions set base Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "base": "EUR",
    },
    "type": "SET_BASE",
  },
]
`;

exports[`main actions set loading Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "loading": true,
    },
    "type": "SET_LOADING",
  },
]
`;

exports[`main actions set remove symbol Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "symbol": "IDR",
    },
    "type": "REMOVE_SYMBOL",
  },
]
`;
